### ISSUE TITLE
Update Travis to calculate code coverage

### ISSUE THREAD

**Issue by arnosthavelka** [arnosthavelka/junit-poc#21]
see:
- documentation: https://docs.travis-ci.com/user/sonarcloud/
- examples:
    - https://github.com/SonarSource/sq-com_example_java-maven-travis
    - https://github.com/cucumber/cucumber-jvm/blob/master/.travis.yml

**Comment by arnosthavelka** [comment]
Try this changes

**pom.xml**
```
    <profiles>
        <profile>
            <id>coverage</id>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.jacoco</groupId>
                        <artifactId>jacoco-maven-plugin</artifactId>
                        <version>0.8.5</version>
                        <executions>
                            <execution>
                                <id>prepare-agent</id>
                                <goals>
                                    <goal>prepare-agent</goal>
                                </goals>
                            </execution>
                            <execution>
                                <id>report</id>
                                <goals>
                                    <goal>report</goal>
                                </goals>
                            </execution>
                        </executions>
                    </plugin>
                </plugins>
            </build>
        </profile>
    </profiles>
```
**.travis.yml**
```
script:
  # the following command line builds the project, runs the tests with coverage and then execute the SonarCloud analysis
  - mvn clean verify sonar:sonar -Pcoverage -Dsonar.projectKey=myorg_myrepo
```
**Others:**

1. update Sonar properties to JDK11
2. remove old sonar project
3. fix sonar key

**Comment by arnosthavelka** [comment]
Sonar includes code coverage now (see https://sonarcloud.io/dashboard?branch=21-update-travis-usage&id=arnosthavelka_junit-poc). However there are some issues to be done:

- add lombok support (to avoid tests for properties)
- fix sonar issues (code coverage and code smells)
- check sonar links (including branch)

**Comment by arnosthavelka** [comment]
fixed by PR #23 -> the rest is moved to #24
